Sources/Validators/CreditCardValidator.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  CreditCardValidator.swift
+//  FormValidatorSwift
+//
+//  Created by Onur Ersel on 2016-11-02.
+//  Copyright © 2016 ustwo. All rights reserved.
+//
+
+import Foundation
+
+/**
+ *  The `CreditCardValidator` contains an `CreditCardCondition`. A valid string is a credit card number.
+ *  - seealso: `CreditCardCondition`
+ */
+public struct CreditCardValidator: Validator {
+    
+    
+    // MARK: - Properties
+    
+    public var conditions: [Condition]
+    
+    
+    // MARK: - Initializers
+    
+    /**
+     Initializes a `CreditCardValidator` with all card types.
+     */
+    public init() {
+        self.init(cardType: .all)
+    }
+    
+    /**
+     Initializes a `CreditCardValidator`.
+     - parameter cardType: Credit card type.
+     */
+    public init(cardType: CreditCardType) {
+        conditions = [CreditCardCondition(cardType: cardType)]
+    }
+    
+    
+    /**
+     Return card types valid with number given.
+    
+     - Parameters:
+      - creditCardNumber: Credit card number.
+      - completion: Completion callback, returns valid card types for given number.
+     */
+    public func cardTypeOf (creditCardNumber: String, completion : @escaping ([CreditCardType])->Void) {
+        
+        
+        DispatchQueue.global().async {
+            
+            var types = [CreditCardType]()
+            defer {
+                DispatchQueue.main.async {
+                    completion(types)
+                }
+            }
+            
+            if self.conditions.count == 0 || !(self.conditions[0] is CreditCardCondition) {
```

Comment:
A couple of points here:
1. I'd recommend using a `guard` statement here, since you're returning from the method.
2. You should use `self.conditions.isEmpty`. It's more efficient than the comparison of the count to zero and reads better.
3. You can put the `let creditCardCondition` statement in here to remove the need for the additional `is` check and it removes the force cast, which is why Travis CI is failing this build.
4. You should probably call the completion handler before calling return even if it's being called with an empty parameter. Otherwise, the developer using the framework could be left hanging waiting for the completion handler to be called.